Automattic/wp-calypso
# Segmentation Survey: Make the question components mapping more flexible

Closes https://github.com/Automattic/dotcom-forge/issues/7412

## Proposed Changes

* Added a `headerAlign` prop to the main SegmentationSurvey component, setting the Entrepreneur flow to a left-aligned survey on the Step level (fixes an issue reported here: https://github.com/Automattic/wp-calypso/pull/91025#pullrequestreview-2074319886)
* Removed the path from the question components mapping. This way, we could add more flexibility in case the developer wants to apply a completely different mapping for the same question types
* Created a `questionConfiguration` prop to define `hideContinue` and `hideSkip` for each question

## Why are these changes being made?

* This PR proposes a more flexible approach to question component mapping. It follows up on the thread I initiated [here](https://github.com/Automattic/wp-calypso/pull/90788#pullrequestreview-2072017187), and the solution is completely open to discussion. I'd appreciate it if folks from @Automattic/vertex could take a look at it, as it also touches on code you recently worked on.

## Testing Instructions

* Apply this PR to your local
* Testing the Guided Flow
  * Go to http://calypso.localhost:3000/start/guided/initial-intent
  * The page header should be center-aligned
  * The first question options should rendered as cards without radio control
  * Continue and skip buttons should be hidden on the first question
  * Clicking on any card should go to the second question (with the continue and skip buttons displayed)
  * The second question option should be rendered as the default checkbox option
* Testing the Entrepreneur Flow
  * Go to http://calypso.localhost:3000/setup/entrepreneur/start
  * The page header should be left-aligned
  * The continue and skip buttons should be displayed for both questions
  * The first question options should be rendered as the default radio control
  * The second question option should be rendered as the default checkbox control
* Perform regression tests to make sure the answers are saved as expected

## Pre-merge Checklist
